### PR TITLE
[DNM] *: remove lifetime of snapshot

### PR DESCRIPTION
**Pull request by BusyJay**
Remove lifetime of snapshot to make code more clean and make it easy to merge snapshot generation into snapshot manager later.

@ngaut @siddontang @disksing PTAL

### CODE REVIEW

File: src/storage/engine/raftkv.rs [hunk 1]
```diff
@@ -99,6 +99,31 @@ pub struct RaftKv<C: PdClient + 'static> {
     router: Arc<RwLock<ServerRaftStoreRouter>>,
 }
 
+enum CmdRes {
+    Resp(Vec<Response>),
+    Snap(RegionSnapshot),
+}
+
+fn on_result(mut resp: RaftCmdResponse, l: usize, uuid: &[u8], db: Arc<DB>) -> Result<CmdRes> {
```
Comment (siddontang) [inline src/storage/engine/raftkv.rs]: s/l/resp_count/

File: tests/storage/test_raftkv.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+use raftstore::server::new_server_cluster;
+
+#[test]
+fn test_raftkv() {
+    let count = 1;
+    let mut cluster = new_server_cluster(0, count);
+    cluster.run();
+
+    // make sure leader have elected.
```
Comment (siddontang) [inline tests/storage/test_raftkv.rs]: have been elected.

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**